New testcases to get more coverage on hera

hugo-dc opened the pull request:
New test cases to get more coverage on hera:

- callValueInStaticMode
- ~createCodeWithRevert~ - create is returning `0x0100000000000000000000000000000000000000000000000000000000000000` instead of `0x01`, check new PR https://github.com/ewasm/tests/pull/68
- getBlockHashOOB
- logZeroLengthMemory
- malformedBytecodeInvalidPreamble
- malformedBytecodeInvalidWasm
- selfDestructExistingAccount
- selfDestructNewAccount

hugo-dc committed:
new testcases to get more coverage on hera

jakelang commented on src/GeneralStateTestsFiller/stEWASMTests/malformedBytecodeInvalidWasmFiller.yml:
```diff
@@ -0,0 +1,51 @@
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      # this bytecode is the same as in getAddress
+      # changed this byte from 10 to 11
```
Perhaps change more than one byte, just to increase the chance Binaryen will throw parseException?

jakelang commented on src/GeneralStateTestsFiller/stEWASMTests/selfDestructExistingAccountFiller.yml:
```diff
@@ -0,0 +1,65 @@
+        (module
+          (import "ethereum" "selfDestruct" (func $selfDestruct (param i32)))
+          (memory 1)
+          (data (i32.const 0) "\80\75\8d\9f\09\8c\e0\89\9d\c0\70\1f\9d\ad\fc\f4\ad\45\8d\c8")
```
I think the beneficiary should be the `a94...0b` address as it is executing the transaction and the hackmd says it should self-destruct back into the original sender.

jakelang commented on src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml:
```diff
@@ -0,0 +1,60 @@
+# selfDestruct new account
+---
+selfDestructNewAccount:
```
I think this was already done, but under a different name. The `selfDestructNewAccount` case seems to be covered by the basic interface test?

jakelang commented on src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml:
```diff
@@ -0,0 +1,60 @@
+      nonce: '0'
+      storage: {}
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
```
Should the balance be 0 here?

hugo-dc committed:
add more invalid bytecodes to malformedBytecodeInvalidWasm test case, fix selfDestruct testcase according to specification

hugo-dc commented:
@jakelang thanks for reviewing, I've changed more bytecodes in the `malformedBytecodeInvalidWasm` test case, also removed `selfDestructNewAccount` testcase because as you said another test case already exists for this I just took the previous test case and changed it according to the spec. Also made changes to `selfDestructNewAccount" to follow the specification.

jakelang reviewed (approved):
Looks good!
